jerry-debugger/src/lib/protocol-handler.ts

```diff
@@ -410,22 +462,46 @@ export class JerryDebugProtocolHandler {
     if (this.delegate.onBreakpointHit) {
       this.delegate.onBreakpointHit(breakpointRef);
     }
+
+    while (true) {
+      const expression = this.evalQueue.shift();
+      if (expression === undefined) {
+        break;
+      }
+      this.evaluate(expression);
+    }
   }
 
-  onBacktrace(message: Uint8Array) {
-    for (let i = 1; i < message.byteLength; i += this.byteConfig.cpointerSize + 4) {
-      const breakpointData = this.decodeMessage('CI', message, i);
+  onBacktrace(data: Uint8Array) {
+    this.logPacket('Backtrace');
+    for (let i = 1; i < data.byteLength; i += this.byteConfig.cpointerSize + 4) {
+      const breakpointData = this.decodeMessage('CI', data, i);
       this.backtrace.push(this.getBreakpoint(breakpointData).breakpoint);
     }
 
-    if (message[0] === SP.JERRY_DEBUGGER_BACKTRACE_END) {
+    if (data[0] === SP.JERRY_DEBUGGER_BACKTRACE_END) {
       if (this.delegate.onBacktrace) {
         this.delegate.onBacktrace(this.backtrace);
       }
+      this.lastBacktrace = this.backtrace;
       this.backtrace = [];
     }
   }
 
+  onEvalResult(data: Uint8Array) {
+    this.logPacket('Eval Result');
+    this.evalResultData = assembleUint8Arrays(this.evalResultData, data);
```

Comment:
Like most of these other JrS debugger packets, a long result is spread over multiple packets: _RESULTs finalized by _RESULT_END. So the half-baked data persists across calls in that case. If there are static variables, I guess it could use that.

Interesting, the .html client at least doesn't handle this right and always treats them as 'final' packets. Unless it is *I* who am mistaken. :)

Reply:
Ah, right, sorry I missed that.